core/sql/optimizer/ValueDesc.cpp

```diff
@@ -3165,7 +3165,11 @@ void ValueIdSet::replaceVEGExpressions
           if (iePtr != exprId.getItemExpr())  // a replacement was done
 	    {
 	      subtractElement(exprId);        // remove existing ValueId
-	      newExpr += iePtr->getValueId(); // replace with a new one
+	      //insert new expression(s)
+	      if (iePtr->getOperatorType() == ITM_AND)
+	    	  iePtr->convertToValueIdSet(newExpr, NULL, ITM_AND, FALSE, FALSE);
```

Comment:
My bad, I provided that fix. What about the following, above line 3170: The replacement of a RangeSpec could be an AND, convert ANDed predicates into additional values in newExpr.